### PR TITLE
Operator docs fixes

### PR DESCRIPTION
**Pull request by nrjpoddar**
* Remove reference to downloading the Istio release as pre-requisite.

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by nrjpoddar** [comment]
@ostromart I think we should also add the following to enhance the docs (I can do that as part of PR if we have an agreement):

* Add profile for demo installation and make that the profile in docs as the first one users should use. The "default" profile seems to have substantial resource requirements which will make it difficult for users to try this out.
* Add demo-auth profile
* Add links to what these profiles are, may be link to Github?

**Comment by nrjpoddar** [comment]
It is also very unclear as to which version of Istio are you installing with the operator.

**Comment by nrjpoddar** [comment]
Makes sense. How does a user easily sets /find out what version of Istio was installed by operator? Currently, I see this image tag for pilot:

`gcr.io/istio-testing/pilot:latest`

**Comment by ostromart** [comment]
Sorry, didn't see your suggestions before making comment above. Yes, I think the first installation that users see should be a demo install. 
If you look at operator.yaml, this points to the default profile, which is the default prod settings. 
We could point it to "profile: demo" which also installs prom + grafana, as well as being lighter on resources (we may want to tweak these a bit). 
We could have demo-auth although is there value in doing strict auth for the demo? Maybe just an default-auth-strict profile, with is default with strict for prod use?
Also, we should probably get rid of the permissive-auth tab and just move the default tab there in its place. I don't think users care much about that this is permissive auth at this point in trying out istio.

File: content/en/docs/setup/install/standalone-operator/index.md [hunk 1]
```diff
@@ -53,7 +52,19 @@ $ kubectl apply -f https://preliminary.istio.io/operator-profile-demo.yaml
...
+
+Run the following command to switch to this profile:
+
+{{< text bash >}}
+$ kubectl apply -f https://preliminary.istio.io/operator-profile-default.yaml
+{{< /text >}}
+
+{{< /tab >}}
+
+{{< tab name="demo auth" cookie-value="demo-auth" >}}
```
Comment (ostromart) [inline content/en/docs/setup/install/standalone-operator/index.md]: maybe call this default strict auth? that's the profile it's pointing to.

**Comment by nrjpoddar** [comment]
@linsun should we keep the demo-auth profile for installer. I think we should keep it as long as we mention it in the non-operator install section. @ostromart and I weren't sure, so thought we would ask you.

File: content/en/docs/setup/install/standalone-operator/index.md [hunk 2]
```diff
@@ -16,44 +16,36 @@ instead, which is a stable feature.
...
-
 1. Perform any necessary [platform-specific setup](/docs/setup/platform-setup/).
 
 1. Check the [Requirements for Pods and Services](/docs/setup/additional-setup/requirements/).
 
 ## Installation steps
 
-1. Install Istio using the operator with the default profile:
+1. Install Istio using the operator with the demo profile. This is only for demo
+   usage and **should not be used in production**.
```
Comment (sdake) [inline content/en/docs/setup/install/standalone-operator/index.md]: I think a callout would be better than this bolded warning.

**Comment by sdake** [comment]
> Sorry, didn't see your suggestions before making comment above. Yes, I think the first installation that users see should be a demo install.
> If you look at operator.yaml, this points to the default profile, which is the default prod settings.
> We could point it to "profile: demo" which also installs prom + grafana, as well as being lighter on resources (we may want to tweak these a bit).
> We could have demo-auth although is there value in doing strict auth for the demo? Maybe just an default-auth-strict profile, with is default with strict for prod use?
> Also, we should probably get rid of the permissive-auth tab and just move the default tab there in its place. I don't think users care much about that this is permissive auth at this point in trying out istio.

Before we completely restructure these docs, I'd like to hear @frankbu 's comments in the docs wg on this topic.  He has taken a look at a related PR, and now that he understands the goals of these pages, he may have some longer term thoughts on how to structure this work.

Cheers
-steve

**Comment by sdake** [comment]
> @linsun should we keep the demo-auth profile for operator. I think we should keep it as long as we mention it in the non-operator install section. @ostromart and I weren't sure, so thought we would ask you.

Master currently has demo as the initially installed profile.  I think demo-auth adds value.  We should document every profile we implement, and longer term for 1.5, the profiles should be tied directly to tasks in the documentation.

cc / @rcaballeromx @frankbu 

@rcaballeromx I'll try to prepare a preliminary plan for installation in 1.5 documentation, but would prefer to defer to you for tidying up the plan and execution if you are willing.


Cheers
-steve

**Comment by nrjpoddar** [comment]
> > @linsun should we keep the demo-auth profile for operator. I think we should keep it as long as we mention it in the non-operator install section. @ostromart and I weren't sure, so thought we would ask you.
> 
> Master currently has demo as the initially installed profile. I think demo-auth adds value. We should document every profile we implement, and longer term for 1.5, the profiles should be tied directly to tasks in the documentation.
> 
> cc / @rcaballeromx @frankbu
> 
> @rcaballeromx I'll try to prepare a preliminary plan for installation in 1.5 documentation, but would prefer to defer to you for tidying up the plan and execution if you are willing.
> 
> Cheers
> -steve

Thanks @sdake for your review. IIUC, I am going to rebase on top of latest master and hopefully most of the changes around demo profile won't be needed for this PR as @linsun already addresses them. I will keep the demo-auth profile and other changes if needed as part of this PR and send it your way again.

**Comment by frankbu** [comment]
I added some questions/concerns about the operator install docs to the doc WG agenda.

https://docs.google.com/document/d/1RGb0NOp0J9QSIrMrZ6wokl16RbSfKHy-6NyTpOQdPa8/edit#

**Comment by ostromart** [comment]
Thanks Frank. The operator docs do need some attention, we can discuss in
the Thu meeting. I read over the agenda and have some thoughts.

**Pull request merged**